polars/polars-lazy/polars-plan/src/logical_plan/builder.rs

```diff
@@ -96,7 +98,46 @@ impl LogicalPlanBuilder {
         .into())
     }
 
-    #[cfg(any(feature = "parquet", feature = "parquet_async"))]
+    #[cfg(all(feature = "parquet", feature = "async"))]
+    #[allow(clippy::too_many_arguments)]
+    pub fn scan_parquet_async(
+        uri: String,
+        n_rows: Option<usize>,
+        cache: bool,
+        parallel: polars_io::parquet::ParallelStrategy,
+        row_count: Option<RowCount>,
+        rechunk: bool,
+        low_memory: bool,
+        cloud_options: Option<CloudOptions>,
+    ) -> BoxFuture<'static, PolarsResult<Self>> {
```

Comment:
I'm guessing that this would also likely be much faster once https://github.com/jorgecarleitao/parquet2/issues/159 is implemented. Parquet stores its metadata in the footer of the file, and the last bytes of the file describe how many bytes the metadata take up. Right now arrow2/parquet2 require a `HEAD` request _first_ to get the content length of the parquet file and _then_ try to load the metadata. But this would be faster by:

1. Fetching the last `N` bytes of a Parquet file using a range request, where `N` is something like 64kb which _often_ includes the entire metadata.
2. Checking the last 4 bytes of the buffer to get the length of the metadata
3. If `N` wasn't enough bytes, do one more fetch to get all the metadata.
4. Then parse the metadata

Maybe I'll find some time to try and make a PR for that in `parquet2`

Reply:
@kylebarron I am working on a native polars parquet reader. Hope we can include this.